### PR TITLE
Refactor BasicImageFieldController

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description
I felt the code was too complex to add the 'rename' feature without introducing technical debt

## Fixes
Related: #5192

## Approach
Refactor the code so it's easier to reason about

## How Has This Been Tested?

On my device - no change

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by david-allison-1** [comment]
Please squash this when merging

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldController.java [hunk 1]
```diff
@@ -400,17 +407,20 @@ private void handleSelectImageIntent(Intent data) {
 
         File internalizedPick = internalizeUri(selectedImage);
         if (internalizedPick == null) {
+            String urlImagePath = getImageInfoFromUri(mActivity, selectedImage).first;
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldController.java:410]: Extracted from getImageUri

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldController.java [hunk 2]
```diff
@@ -643,22 +662,35 @@ private void handleCropResult() {
         Timber.d("handleCropResult");
         if (!rotateAndCompress()) {
             Timber.i("handleCropResult() appears to have an invalid file, reverting");
-            revertToPreviousImage();
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/BasicImageFieldController.java:646]: Handled in `rotateAndCompress` now

**Review by mikehardy — APPROVED** [review]
You requested a squash merge, and the PR took a conflict from unrelated lint refactors so I just fixed them (I hope!) using the web UI here which produces a merge commit but with squash merge that doesn't matter. Hopefully it passes CI and we're just done with no further action here

Regardless: THANK YOU for cleaning this object up, what a mess it was!

**Pull request merged**